### PR TITLE
CirrusCI: Bump Ubuntu image to 18.04

### PR DESCRIPTION
**Pull request by ibuclaw**
Assessing the damage and how long we may have to wait until enough releases have occurred for it to be trusted.

The assumption is that the first capable release of supporting Ubuntu 18.04 on all architectures DMD supports will be 2.097.2.

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]

**Comment by ibuclaw** [comment]
If it is 2.097.2, then I guess this will remain a draft for at least 6 months. ;-)

**Comment by ibuclaw** [comment]
- Have tried v2.079.0, v2.090.0, and v2.097.0, all fail on the x86 bootstrap, so have set it to v2.097.1.
- x86_64 bootstrap is still v2.079.0, so I guess it isn't as bad as I thought it might be.
- Re-added LDC (latest) pipeline.
- Experimentally testing GDC-9 _and_ GDC-10 in separate pipelines.

File: ci.sh [hunk 1]
```diff
@@ -26,7 +26,7 @@ build_path=generated/$OS_NAME/release/$MODEL
 if [ "$OS_NAME" == "linux" ]; then
     mkdir -p linker
     rm -f linker/ld
-    ln -s /usr/bin/ld.gold linker/ld
+    ln -s /usr/bin/ld.bfd linker/ld
```
Comment (kinke) [inline ci.sh:29]: ?

File: cirrusci.sh [hunk 2]
```diff
@@ -15,10 +15,12 @@ if [ ! -z ${HOST_DC+x} ] ; then HOST_DMD=${HOST_DC}; fi
 if [ -z ${HOST_DMD+x} ] ; then echo "Variable 'HOST_DMD' needs to be set."; exit 1; fi
 
 if [ "$OS_NAME" == "linux" ]; then
-  packages="git-core make g++ gdb curl libcurl3 tzdata zip unzip xz-utils"
+  # Needed to bootstrap installation of tzdata for ubuntu 18.04 images
+  ln -snf /usr/share/zoneinfo/UTC /etc/localtime && echo UTC > /etc/timezone
```
Comment (kinke) [inline cirrusci.sh:19]: This isn't needed for LDC CI, which also installs `tzdata`: https://github.com/ldc-developers/ldc/blob/dae61cb8d745642e9eebc85c294c93d2f62c07d9/.cirrus.yml#L58-L72

Comment (ibuclaw) [reply cirrusci.sh:19]: Probably `export DEBIAN_FRONTEND=noninteractive` does it for you.

Comment (kinke) [reply cirrusci.sh:19]: What is/was needed for gdmd only: https://github.com/ldc-developers/ldc/blob/dae61cb8d745642e9eebc85c294c93d2f62c07d9/.cirrus.yml#L144-L145

File: ci.sh (hunk 1 above)
Comment (ibuclaw) [reply ci.sh:29]: GOLD doesn't work on PIE-x86, there are a number of Debian (and probably Ubuntu too) bug reports for it.

Comment (kinke) [reply ci.sh:29]: LDC defaults to gold on Linux for a while, incl. linking 32-bit stuff, without any problems I would remember.

Comment (ibuclaw) [reply ci.sh:29]: For example: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=846921
Also, there's a dmd issue: https://issues.dlang.org/show_bug.cgi?id=18798
There's an acknowledged issue with gold in binutils: https://sourceware.org/bugzilla/show_bug.cgi?id=18296

Comment (kinke) [reply ci.sh:29]: And if it really doesn't work for DMD somehow, I'd suggest skipping this block in that case only - `if [[ "$OS_NAME" == "linux" && "$MODEL" != "32" ]]`.

File: cirrusci.sh (hunk 2 above)
Comment (ibuclaw) [reply cirrusci.sh:19]: > What is/was needed for gdmd only: https://github.com/ldc-developers/ldc/blob/dae61cb8d745642e9eebc85c294c93d2f62c07d9/.cirrus.yml#L144-L145

You can blame that on Perl. ;-)

**Comment by ibuclaw** [comment]
Someone should really add @kinke and I to the TeamCI team (assuming that team has access to the protected branch/pull request settings of dmd) with the departing absence of Seb.

**Comment by ljmf00** [comment]
I made a similar PR to do this https://github.com/dlang/dmd/pull/12822 .

**Review by kinke — APPROVED** [review]

File: ci.sh (hunk 1 above)
Comment (Geod24) [reply ci.sh]: > LDC defaults to gold on Linux for a while, incl. linking 32-bit stuff, without any problems I would remember.

FYI I had to change it back to bfd on Alpine as well.

**Comment by Geod24** [comment]
> Someone should really [...]

Did something to that effect. Feel free to ping me for org admin stuff, be it for `dlang` or `dlang-community`.

**Review by Geod24 — APPROVED** [review]
Thanks!

**Comment by ibuclaw** [comment]
> I made a similar PR to do this #12822 .

Can you split up each platform?  Thanks!

**Comment by ljmf00** [comment]
> > I made a similar PR to do this #12822 .
> 
> Can you split up each platform? Thanks!

Sure, I'll do it :)

**Pull request merged**